src/Dict.h

```diff
@@ -120,6 +125,10 @@ class DictEntry {
 		}
 
 	const char* GetKey() const { return key_size <= 8 ? key_here : key; }
+	detail::HashKey* GetHashKey() const { return new detail::HashKey(GetKey(), key_size, hash); }
```

Comment:
I was looking at those HashKey constructors the other day and wondered why we have three constructors that all do roughly the same things, but with different arguments. It seemed like we could just have `HashKey(void* arg_key, int arg_size, bool copy_key)` and `HashKey(void* arg_key, int arg_size, hash_t arg_hash, bool copy_key)`.

The context for the above is that I was playing with implementing structured binding support for DictEntry, where DictEntry could just store a HashKey internally and return a reference there. It gets really sticky though because we use `realloc` in the Dictionary code, and it doesn't like realloc'ing those objects very much.

I agree with just returning an object here though instead of a pointer. It saves the concern of not deleting the returned value, and has the nice side effect of fixing all of those false-positive Coverity findings around them.

Reply:
I went ahead and made it a `unique_ptr`. We do use the result as a pointer in `CompHash::SingleValHash` and `TableVal::AddTo` for varying reasons. The former can easily be converted to take an object, but the latter is a bit stickier without greater changes (or copying the key data a second time, and then later a third time).